### DISCUSSION TITLE
How do I update TagStudio in a risk-free way?

### DISCUSSION THREAD

**Discussion by ParDiv01** [TagStudioDev/TagStudio#324] [Q&A]
Since TagStuio is still in alpha, is it possible to lose your tag data? If so, how can I reduce the possibility of losing my data, and how do I backup my data? Thank you for the great app

**Comment by CyanVoxel** [comment]
Thank you for checking out the program! TagStudio is designed to keep your save data separate from the program itself in the following ways:
1. Library data is stored within a `.TagStudio` folder inside each top-level folder you open as a library. This means that the data is portable across computers and installations of TagStudio, so the program can be uninstalled or updated without touching this data.
2. User preferences are stored in a standard location, for example at `C:/Users/<User>/AppData/Roaming/TagStudio/TagStudio.ini` on Windows by default. This is further customizable via a launch argument.

For backing up data, TagStudio has a "Save Library Backup" feature in the file menu that stores a copy of your `ts_library.json` file inside the `.TagStudio/backups` folder for your library. Currently there's no automatic way to restore from a backup, and you'll need to manually move and rename these backup files to replace your main `.TagStudio/ts_library.json` file. I would recommend making periodic backups in this way, and even storing some outside of the `.TagStudio` folder itself.
![image](https://github.com/user-attachments/assets/20eb1653-bb00-473b-8677-488efe85ed14)
![image](https://github.com/user-attachments/assets/d9978d5a-1d01-4770-8627-b38cea660702)


We do our best to make sure that data is preserved and not corrupted though updates, however it's impossible to give a 100% guarantee. Backups are the only guaranteed way to prevent against data loss and corruption.
